apps/server/src/modules/school/school.module.ts

```diff
@@ -0,0 +1,10 @@
+import { Module } from '@nestjs/common';
+import { SchoolRepo } from '@shared/repo';
+import { SchoolUc } from '@src/modules/school/uc/school.uc';
+import { SchoolService } from '@src/modules/school/service/school.service';
+
+@Module({
+	providers: [SchoolRepo, SchoolUc, SchoolService],
+	exports: [SchoolUc, SchoolService],
```

Comment:
Nest says we do:

    Nest can't resolve dependencies of the SchoolUc (?). Please make sure that the argument SchoolService at index [0] is available in the ProvisioningModule context.

    Potential solutions:
    - If SchoolService is a provider, is it part of the current ProvisioningModule?
    - If SchoolService is exported from a separate @Module, is that module imported within ProvisioningModule?
      @Module({
        imports: [ /* the Module containing SchoolService */ ]
      })

Reply:
This happens because you added the SchoolUc to the ProvisioningModule's providers. So it tries to create a new instance of SchoolUc for ProvisioningModule. So it needs all the dependencies to be available in ProvisioningModule as well.

If you remove SchoolUc from the providers array, the ProvisioningModule will use the SchoolUc instance provided and expoted by the SchoolModule.

Reply:
Thanks. We will remove this when we implement the sanis strategy in the next sprint